src/lib.rs

```diff
@@ -115,9 +117,9 @@ impl EncodingDetector {
     //    unsafe { ffi::uchardet_reset(self.ptr); }
     //}
 
-    /// Get the decoder's current best guess as to the encoding.  Returns
+    /// Get the decoder's current best guess as to the encoding. Returns
     /// `None` on error, or if the data appears to be ASCII.
-    fn charset(&self) -> Option<String> {
+    fn charset(&self) -> Result<String, EncodingDetectorError> {
```

Comment:
I'd like to try the conversion. Though it's my first time using `quick-error` so tell me if I'm doing dumb stuff.